File: app/src/main/java/kr/purplebeen/half/MainActivity.java
package kr.purplebeen.half;

import androidx.appcompat.app.AppCompatActivity;
import io.socket.client.IO;
import io.socket.client.Socket;
import io.socket.emitter.Emitter;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import java.net.URISyntaxException;

public class MainActivity extends AppCompatActivity {
    private TextView resultText;
    private Socket socket;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        startActivity(new Intent(this, DeviceRegisterActivity.class));

        init();
        try {
            Log.d("asdf", "asdf");
            setSocket();
        } catch (URISyntaxException e) {
            e.printStackTrace();
        }

    }

    private void init() {
        resultText = findViewById(R.id.resultText);
    }

    private void setSocket() throws URISyntaxException {
        Log.e("asdf2", "asdf");
        socket = IO.socket("http://10.27.2.150:3001");
        socket.on("result", (Emitter.Listener) args -> {
            runOnUiThread(() -> {
                resultText.setText(args[0].toString());
                Log.d("asdf", args[0].toString());
            });
        });
        socket.connect();
    }
}

File: app/src/main/java/kr/purplebeen/half/MainActivity.kt
package kr.purplebeen.half

import androidx.appcompat.app.AppCompatActivity
import io.socket.client.IO
import io.socket.client.Socket
import io.socket.emitter.Emitter

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

import java.net.URISyntaxException

class MainActivity : AppCompatActivity() {
    private var socket: Socket? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        try {
            setSocket()
        } catch (e: URISyntaxException) {
            e.printStackTrace()
        }

    }

    @Throws(URISyntaxException::class)
    private fun setSocket() {
        socket = IO.socket("http://10.27.2.150:3001")
        socket!!.on("result") {
            runOnUiThread {
                Thread.sleep(1000)
                if(it[0].toString() == "dog_bark") {
                    resultImage.setBackgroundResource(R.drawable.dogbark)
                } else if(it[0].toString() == "gun_shot") {
                    resultImage.setBackgroundResource(R.drawable.gunshot)
                } else if(it[0].toString() == "siren") {
                    resultImage.setBackgroundResource(R.drawable.siren)
                } else if(it[0].toString() == "car_horn") {
                    resultImage.setBackgroundResource(R.drawable.carhorn)
                }
                Log.d("asdf", it.toString())
            }

        }
        socket!!.connect()
    }
}
